app/scripts/metamask-controller.js

```diff
@@ -2936,6 +2959,48 @@ export default class MetamaskController extends EventEmitter {
         },
         getAuthorizedScopesByOrigin,
       );
+      if (process.env.MULTICHAIN_API) {
+        this.controllerMessenger.subscribe(
+          `${this.accountsController.name}:selectedAccountChange`,
+          async (account) => {
+            if (
+              account.type === 'solana:data-account' &&
+              account.address !== lastSelectedSolanaAccountAddress
+            ) {
+              lastSelectedSolanaAccountAddress = account.address;
+
+              const originsWithSolanaAccountChangedNotifications =
+                getOriginsWithSessionProperty(
+                  this.permissionController.state,
+                  'solana_accountChanged_notifications',
+                );
+
+              const solanaAccounts = getPermittedAccountsForScopesByOrigin(
+                this.permissionController.state,
+                [
+                  MultichainNetworks.SOLANA,
+                  MultichainNetworks.SOLANA_DEVNET,
+                  MultichainNetworks.SOLANA_TESTNET,
```

Comment:
thoughts on targeting by namespace? Not saying we necessarily make that change, but it might be more aligned with what we're trying to achieve here

Reply:
agreed. Unfortunately this will require a change to (or a new helper in core called `getPermittedAccountsForNamespace`) Perhaps we could leave this for a followup?